authorization/src/main/java/io/deephaven/auth/codegen/impl/ApplicationServiceAuthWiring.java

```diff
@@ -0,0 +1,37 @@
+/**
+ * Copyright (c) 2016-2022 Deephaven Data Labs and Patent Pending
+ */
+/**
+ * ---------------------------------------------------------------------------------------------------------------------
+ * AUTO-GENERATED CLASS - DO NOT EDIT MANUALLY - for any changes edit GenerateServiceAuthWiring and regenerate
+ * ---------------------------------------------------------------------------------------------------------------------
+ */
+package io.deephaven.auth.codegen.impl;
+
+import com.google.rpc.Code;
+import io.deephaven.auth.AuthContext;
+import io.deephaven.auth.ServiceAuthWiring;
+import io.deephaven.proto.util.Exceptions;
+
+public interface ApplicationServiceAuthWiring extends ServiceAuthWiring {
+
+    void checkPermissionListFields(
+            AuthContext authContext,
+            io.deephaven.proto.backplane.grpc.ListFieldsRequest request);
+
+    class AllowAll implements ApplicationServiceAuthWiring {
+
+        public void checkPermissionListFields(
+                AuthContext authContext,
+                io.deephaven.proto.backplane.grpc.ListFieldsRequest request) {}
+    }
+
+    class DenyAll implements ApplicationServiceAuthWiring {
+
+        public void checkPermissionListFields(
+                AuthContext authContext,
+                io.deephaven.proto.backplane.grpc.ListFieldsRequest request) {
+            throw Exceptions.statusRuntimeException(Code.PERMISSION_DENIED, "Operation not allowed.");
+        }
+    }
```

Comment:
Do we imagine implementations might want to extend from any of the AllowAll or DenyAll classes? While it might be "convenient", I think we should force implementations to always implement each method themselves (forces a compile error on new releases instead of opting into potentially a bad choice).

On a related note, these could all be implemented as enum singletons.

Reply:
Hmm. I would rather place a disclaimer and offer recommended practices rather than force the user to copy paste. An extension of the DenyAll actually feels rather appropriate for the scenario where you allow a smaller subset of functionality.

I'll discuss with Ryan.